.github/workflows/smoketest.yml

```diff
@@ -56,3 +65,23 @@ jobs:
         run:  |
           chmod +x smoke_tester
           ./smoke_tester test --dut-definitions $SMOKE_TESTER_CONFIG --markdown-summary $GITHUB_STEP_SUMMARY
+
+  embedded-test:
+    runs-on: ["self-hosted", "linux", "ARM64", "${{ matrix.runner }}"]
+    strategy:
+      matrix:
+        runner: ["smoke-tester-2"]
+
+    needs: build
+
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: probe-rs
+
+      - name: Run embedded-tests
```

Comment:
What tests are we running though?

Reply:
whatever is compiled for the target. the point is not the HIL test, but our ability to execute those tests. this is meant to test flashing and semihosting at this moment.

The C2 test I have right now asserts that 1+1=2

Reply:
As a side effect, we now have essentially nightly builds of probe-rs for raspberry pis :D